File: ethstats/src/main/java/org/apache/tuweni/ethstats/TxStats.java
package org.apache.tuweni.ethstats;

import com.fasterxml.jackson.annotation.JsonGetter;
import org.apache.tuweni.crypto.Hash;

final class TxStats {

    private final Hash hash;

    TxStats(Hash hash) {
        this.hash = hash;
    }

    @JsonGetter("hash")
    public Hash getHash() {
        return hash;
    }
}

File: ethstats/src/main/kotlin/org/apache/tuweni/ethstats/TxStats.kt
// Copyright The Tuweni Authors
// SPDX-License-Identifier: Apache-2.0
package org.apache.tuweni.ethstats

import org.apache.tuweni.eth.Hash

/**
 * Stats reported to ethnetstats representing the hash of a transaction.
 */
data class TxStats(val hash: Hash)
